packages/issuer/lib/VcIssuer.ts

```diff
@@ -98,10 +101,9 @@ export class VcIssuer {
       }
 
       const { iss, aud, iat, nonce } = payload
-      // Get client id and check if it is pre-authorized
-      if (iss && iss !== '') {
-        throw new Error(ISS_ERROR)
-      }
+      // I couldn't find a way to get the client_id, neither the grant type at this point, so I will not validate iss since it's optional
+      // iss: OPTIONAL (string). The value of this claim MUST be the client_id of the client making the credential request.
+      // This claim MUST be omitted if the Access Token authorizing the issuance call was obtained from a Pre-Authorized Code Flow through anonymous access to the Token Endpoint.
```

Comment:
You should know whether you are in an authorized or pre-authorized context. In other words whether it should be present or not

Reply:
I could know from the jwt, if iss is present, it's authorized flow. However, I'm validating the jwt in the credential endpoint and the only thing I have is the jwt, I don't have anything to validate the jwt against. The only thing I can think of is an introspection endpoint in the AS.

Reply:
You could use the state manager to get the credential offer grants. If only one is present that has to be used. If 2 were present then indeed you cannot know whether it should be present or not

Reply:
Fixed